examples/timeSeries-hand-gestures/sketch.js

```diff
@@ -0,0 +1,192 @@
+
+let handPose;
+let video;
+
+let hands = [];
+let sequence = [];
+
+let recordingFinished = false;
+let predictedWord = "";
+
+// UI variables
+let trainingWords = {};
+
+function preload() {
+  // Load the handPose model
+  handPose = ml5.handPose();
+
+  // setup the timeseries neural network
+  let options = {
+    outputs: ["label"],
+    task: "classification",
+    dataMode: "spatial",
+    debug: "true",
+    learningRate: 0.001,
+  };
+  model = ml5.timeSeries(options);
+}
+
+function setup() {
+  createCanvas(640, 480);
+
+  // setup video capture
+  video = createCapture(VIDEO);
+  video.size(640, 480);
+  video.hide();
+
+  // place UI elements
+  UI();
+
+  // use handpose model on video
+  handPose.detectStart(video, gotHands);
+}
+
+function draw() {
+  // draw video on frame
+  image(video, 0, 0, width, height);
+
+  drawPredictedWord();
+
+  // if hands are found then start recording
+  if (hands.length > 0 && recordingFinished == false) {
+    if (sequence.length <= seqLength) {
+      // get coordinates from hands (21 points)
+      handpoints = drawPoints();
+      sequence.push(handpoints);
+
+      // once sequence reaches the seqLength, add sequence as just one X value
+    } else if (sequence.length > 0) {
+      // get the training word from the input box
+      let train_word = nameField.value();
+
+      // if there is a word currently in the box then add data with that label
+      if (train_word.length > 0) {
+        // add data to the model
+        let target = { label: train_word };
+        model.addData(sequence, target);
+        trainingWordsUpdate();
+
+        // if there is no word in the box then classify instead
+      } else {
+        // classify the data
+        model.classify(sequence, gotResults);
+      }
+
+      // reset the sequence
+      sequence = [];
+      recordingFinished = true;
+    }
+
+    // can only record again when hand is out of frame
+  } else {
+    if (hands.length == 0) {
+      recordingFinished = false;
+    }
+  }
+}
+
+function drawPoints() {
+  let handpoints = [];
+  // iterate through both hands
+  for (let i = 0; i < hands.length; i++) {
+    let hand = hands[i];
+    for (let j = 0; j < hand.keypoints.length; j++) {
+      // access the keypoints in the hand
+      let keypoint = hand.keypoints[j];
+      handpoints.push(keypoint.x, keypoint.y);
+
+      fill(0, 255, 0);
+      noStroke();
+      circle(keypoint.x, keypoint.y, 5);
+    }
+  }
+  // assign to a different variable before clearing
+  let output = handpoints;
+  handpoints = [];
+
+  return output;
+}
+
+// Callback function for when handPose outputs data
+function gotHands(results) {
+  // save the output to the hands variable
+  hands = results;
+}
+
+function trainModelAndSave() {
+  model.normalizeData();
+  let options = {
+    epochs: 100,
+  };
+  model.train(options, whileTraining, finishedTraining);
+  nameField.value("");
+}
+
+function whileTraining(epoch) {
+  console.log(epoch);
```

Comment:
Easy to get confused with console.log()s that aren't your (the user's) own... :)

Best to make them as descriptive as possible (e.g. `console.log("In epoch", epoch);` and then commenting them out to, so that they're still available for future debugging.